metrics/ci_integrations/test/cli/logger/logger_test.dart

```diff
@@ -1,65 +1,197 @@
 import 'package:test/test.dart';
 
-import '../test_util/mock/io_sink_mock.dart';
+import '../test_util/mock/logger_factory_mock.dart';
+import '../test_util/mock/logger_writer_mock.dart';
 
 void main() {
   group("Logger", () {
-    final sinkMock = IOSinkMock();
+    const testSourceClass = Object;
+    const testVerbose = true;
+    final writerMock = LoggerWriterMock();
 
-    const message = 'message';
-    const error = 'error';
+    final logger = Logger(
+      writer: writerMock,
+      verbose: false,
+      sourceClass: testSourceClass,
+    );
+
+    final verboseLogger = Logger(
+      writer: writerMock,
+      verbose: testVerbose,
+      sourceClass: testSourceClass,
+    );
+
+    final loggerFactoryMock = LoggerFactoryMock();
+
+    setUpAll(() {
+      LoggerManager.setLoggerFactory(loggerFactoryMock);
+    });
 
     tearDown(() {
-      reset(sinkMock);
+      reset(writerMock);
+      reset(loggerFactoryMock);
+      LoggerManager.instance.reset();
     });
 
-    test(".setup() sets up the Logger", () {
-      Logger.setup(messageSink: sinkMock);
+    tearDownAll(() {
+      LoggerManager.setLoggerFactory(null);
+    });
+
+    bool matchesPrefixes(String log, String sourceClass, String message) {
+      final regexp = RegExp(r'(\d+\/?)+\ (\d{2}:?)');
+      final timestampEnd = regexp.firstMatch(log).end;
+      return log.contains(sourceClass, timestampEnd) && log.endsWith(message);
+    }
 
-      expect(() => Logger.logMessage(message), returnsNormally);
+    test("throws an ArgumentError if the given source class is null", () {
+      expect(
+        () => Logger(
+          sourceClass: null,
+          writer: writerMock,
+          verbose: testVerbose,
+        ),
+        throwsArgumentError,
+      );
     });
 
-    test(".logError() logs the given error to the error sink", () {
-      Logger.setup(errorSink: sinkMock);
-      Logger.logError(error);
+    test("throws an ArgumentError if the given writer is null", () {
+      expect(
+        () => Logger(
+          sourceClass: testSourceClass,
+          writer: null,
+          verbose: testVerbose,
+        ),
+        throwsArgumentError,
+      );
+    });
+
+    test("throws an ArgumentError if the given source class is null", () {
+      expect(
+        () => Logger(
+          sourceClass: testSourceClass,
+          writer: writerMock,
+          verbose: null,
+        ),
+        throwsArgumentError,
+      );
+    });
 
-      verify(sinkMock.writeln(error)).called(1);
+    test("creates a new instance with the given parameters", () {
+      expect(
+        () => Logger(
+          sourceClass: testSourceClass,
+          writer: writerMock,
+          verbose: testVerbose,
+        ),
+        returnsNormally,
+      );
     });
 
     test(
-      ".logMessage() logs the given message to the message sink",
+      ".forClass() returns a new logger if LoggerManager does not contain any for the given class type",
```

Comment:
Maybe we can just test that this method delegates to the `LoggerManager.getLogger()`? It seems like we've already tested that the `getLogger()` works correctly in `LoggerManager` tests.

Reply:
Resolve this as we discussed via chat